types/koa/index.d.ts

```diff
@@ -516,7 +516,7 @@ declare class Application<
      * Return a request handler callback
      * for node's native http/http2 server.
      */
-    callback(): (req: IncomingMessage | Http2ServerRequest, res: ServerResponse | Http2ServerResponse) => void;
+    callback(): (req: IncomingMessage | Http2ServerRequest, res: ServerResponse | Http2ServerResponse) => Promise<void>;
```

Comment:
Looking at the original implementation this method would return the result of `this.handleRequest(ctx, fn)`, which returns a promise `fnMiddleware(ctx).then(handleResponse).catch(onerror)`, so seems to be a legit change...